src/containers/explore/components/EntityDataGrid.js

```diff
@@ -0,0 +1,145 @@
+import { faSortDown } from '@fortawesome/pro-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { List, Map } from 'immutable';
+import {
+  Colors,
+  IconButton,
+  Label,
+  Typography,
+} from 'lattice-ui-kit';
+import { DataUtils } from 'lattice-utils';
+import { useSelector } from 'react-redux';
+import type { FQN, UUID } from 'lattice';
+
+import { Flip } from '../../../components';
+import { FQNS } from '../../../core/edm/constants';
+import { selectOrgDataSetColumns } from '../../../core/redux/selectors';
+
+const { NEUTRAL, PURPLE } = Colors;
+const { getPropertyValue } = DataUtils;
+
+const downIcon = <FontAwesomeIcon icon={faSortDown} size="xs" />;
+
+const DataGrid = styled.div`
+  display: grid;
+  flex: 1;
+  font-size: 14px;
+  grid-gap: 20px 30px;
+  grid-template-columns: repeat(auto-fill, minmax(200px, auto));
+
+  > div {
+    max-width: 500px;
+    word-break: break-word;
+  }
+`;
+
+const CountText = styled.div`
+  color: ${PURPLE.P300};
+  font-weight: 600;
+  margin: 0 5px;
+`;
+
+const InfoText = styled.div`
+  align-items: center;
+  color: ${NEUTRAL.N400};
+  display: flex;
+  margin: 15px 0;
+
+  button {
+    margin-left: 5px;
+  }
+`;
+
+const ValueList = styled.ul`
+  list-style: ${({ size }) => (size > 1 ? 'disc' : 'none')} inside;
+  margin: 0;
+  padding-left: 2px;
+`;
+
+const ValueListItem = styled.li`
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+`;
+
+type Props = {
+  data :Object;
+  dataSetId :UUID;
+  organizationId :UUID;
+};
+
+const EntityDataGrid = ({ data, dataSetId, organizationId } :Props) => {
+  const [showFull, setShowFull] = useState(false);
+
+  const dataSetColumns :List<Map<FQN, List>> = useSelector(
+    selectOrgDataSetColumns(organizationId, dataSetId)
+  );
+
+  const labels = {};
+  dataSetColumns.forEach((column) => {
+    const type = getPropertyValue(column, [FQNS.OL_TYPE, 0]);
+    const title = getPropertyValue(column, [FQNS.OL_TITLE, 0]);
+    labels[type] = title;
+  });
+
+  const expandFields = () => {
+    setShowFull(!showFull);
+  };
+
+  const items = [];
+  dataSetColumns.forEach((column :Map) => {
+    const type = getPropertyValue(column, [FQNS.OL_TYPE, 0]);
+    const title = getPropertyValue(column, [FQNS.OL_TITLE, 0]);
+    const values :List = data.get(type, List());
+
+    const elements = [];
+    if (!values.isEmpty()) {
+      values.forEach((value, index) => {
+        elements.push(
+          <ValueListItem key={index.toString()}>{value}</ValueListItem>
+        );
+      });
+    }
+
+    if (elements.length) {
+      items.push(
+        <div key={type}>
+          <Label subtle>{title}</Label>
+          <ValueList size={elements.length}>{elements}</ValueList>
+        </div>
+      );
+    }
+  });
+
+  return (
+    <>
+      <DataGrid>
+        { items.slice(0, 8) }
+        { showFull && items.slice(8) }
+      </DataGrid>
+      {
+        !!items.slice(8).length && (
```

Comment:
A thought on this: `!!items.slice(8).length`. I'm sure this works but it's not very intuitive / immediately obvious, which I think is important when reading conditions like this. In addition, the sole purpose of this expression is to evaluate a boolean and I think this is a wasteful approach since `slice()` will create a copy of the array which just gets thrown away. Finally, you're already doing `items.slice(8)` above; why not reuse it?

I think the following is cleaner 

```
const hiddenItems = items.splice(8);

return (
  <>
    <DataGrid>
      {items}
      {showFull && hiddenItems}
    </DataGrid>
    {
      hiddenItems.length > 0 && (
```